sr_hand_health_report/src/sr_hand_health_report/position_sensor_noise_check.py

```diff
@@ -15,40 +15,76 @@
+        """
         super().__init__(hand_side, fingers_to_test)
-        self._check_duration = rospy.Duration(5.0)
+        self._name = "Position Sensor Noise"
+        self._check_duration = rospy.Duration(3.0)
         self._shared_dict = {}
         self._publishing_rate = rospy.Rate(200)
         self._initial_raw_value = None
+        self._result = {'position_sensor_noise': {}}
 
     def run_check(self):
-        result = {"position_sensor_noise_check": []}
+        """
+            Runs the test for selected fingers.
+            @return: Dictionary with the results of the check
+        """
+        self._result = {'position_sensor_noise': {}}
+        if self._stopped_execution:
+            self._stopped_execution = False
+            return
         rospy.loginfo("Running Position Sensor Noise Check")
-        rospy.sleep(3.0)
+        result = dict(self._result)
 
         for finger in self.fingers_to_check:
-            rospy.loginfo("collecting and analyzing data for FINGER {}".format(finger.finger_name))
+            rospy.loginfo("collecting and analyzing data for finger {}".format(finger.finger_name))
             for joint in finger.joints_dict.values():
-                rospy.loginfo("collecting and analyzing data for JOINT {}".format(joint.joint_name))
+                rospy.loginfo("collecting and analyzing data for joint {}".format(joint.joint_name))
                 self._initial_raw_value = joint.get_raw_sensor_data()
                 self.check_joint_raw_sensor_value(self._initial_raw_value, joint, self._shared_dict)
-        result["position_sensor_noise_check"].append(dict(self._shared_dict))
+
+                if self._stopped_execution:
+                    break
+            if self._stopped_execution:
+                self._stopped_execution = False
+                return
+
+        result["position_sensor_noise"].update(dict(self._shared_dict))
         rospy.loginfo("Position Sensor Noise Check finished, exporting results")
-        return result
+        self._result = result
+        self._stopped_execution = True
+        return
 
     def check_joint_raw_sensor_value(self, initial_raw_value, joint, dictionary):
+        """
+            Checks the sensor noise and saves the result to _shared_dict
+            @param initial_raw_value: Float value of raw sensor reading
+            @param Joint: Joint object being under test
+            @param dictionary: Dictionary where the result is saved
+        """
         status = ""
         test_failed = False
         if joint.joint_name != self._hand_prefix + "_wrj1" and joint.joint_name != self._hand_prefix + "_thj5":
             initial_raw_value = initial_raw_value[-1:]
         for index in range(len(initial_raw_value)):
             time = rospy.Time.now() + self._check_duration
             while rospy.Time.now() < time and test_failed is not True:
+                if self._stopped_execution:
+                    break
```

Comment:
return instead of break?

Reply:
In the comment above, I explained the sense behind using breaks and not returns. Have a look and if you're not happy lets discuss :)